satpy/readers/fci_l1c_nc.py

```diff
@@ -700,6 +709,31 @@ def calibrate_rad_to_refl(self, radiance, key):
         res = 100 * radiance * np.float32(np.pi) * np.float32(sun_earth_distance) ** np.float32(2) / cesi
         return res
 
+    def _collect_listed_variables(self, file_handle, listed_variables, filetype_info):
+        listed_variables = self._recycle_shared_info(listed_variables, filetype_info)
+        super()._collect_listed_variables(file_handle, listed_variables, filetype_info)
+        self._store_shared_info(filetype_info)
+
+    def _recycle_shared_info(self, listed_variables, filetype_info):
+        if "shared_info" in filetype_info:
+            shared_info = filetype_info["shared_info"]
+            for key in shared_info:
+                self.file_content[key] = shared_info[key]
+                try:
+                    listed_variables.remove(key)
+                except ValueError:
+                    pass
+        return listed_variables
+
+    def _store_shared_info(self, filetype_info):
+        if "shared_info" not in filetype_info:
+            shared_info = {}
+            for key in self.file_content:
+                if any(key.endswith(k) for k in NONSHAREABLE_VARIABLE_ENDINGS):
+                    continue
```

Comment:
By default you are sharing (in case of unknown variable names).  Would it be safer to default to not-sharing, i.e. use a whitelist rather than a blacklist?

Reply:
I think your dict of sharing stuff would make this obsolete in any case as it explicitly says what can be shared and how.